docs/docset.yml

```
@@ -22,7 +22,17 @@ cross_links:
   - kibana
   - logstash
 toc:
-  - toc: reference
+  - file: reference/index.md
+    children:
+    - toc: reference/elasticsearch
+    - toc: reference/community-contributed
+    - toc: reference/enrich-processor
+    - toc: reference/search-connectors
+    - toc: reference/elasticsearch-plugins
+    - toc: reference/query-languages
+    - toc: reference/scripting-languages
+    - toc: reference/text-analysis
+    - toc: reference/aggregations
```

```suggestion
  - toc: reference
   
```

with `reference/toc.yml` holding:

```yaml
 children:
    - toc: elasticsearch
    - toc: community-contributed
    - toc: enrich-processor
    - toc: search-connectors
    - toc: elasticsearch-plugins
    - toc: query-languages
    - toc: scripting-languages
    - toc: text-analysis
    - toc: aggregations
```

Otherwise it will trip the new validations: https://github.com/elastic/docs-builder/pull/767

---

I just did this for now https://github.com/elastic/elasticsearch/pull/125118/commits/1b4df489313fad612b786831cb7dc3324445b8ac.

---

there is no restriction on files origin, just depth so toc.yml file can hold toc.yml files as long as `max_toc_depth` is adhered too.